src/frontend-common/common_host_interface.cpp

```diff
@@ -3344,6 +3353,43 @@ void CommonHostInterface::ReloadPostProcessingShaders()
     AddOSDMessage(TranslateStdString("OSDMessage", "Post-processing shaders reloaded."), 10.0f);
 }
 
+void CommonHostInterface::ToggleWidescreen()
+{
+  g_settings.gpu_widescreen_hack = !g_settings.gpu_widescreen_hack;
+  const GameSettings::Entry* gs = m_game_list->GetGameSettings(System::GetRunningPath(), System::GetRunningCode());
+  DisplayAspectRatio userRatio;
+  if (gs && gs->display_aspect_ratio.has_value())
+    userRatio = gs->display_aspect_ratio.value();
+  else
+  {
+    Settings o_settings;
```

Comment:
Doesn't need to go through the whole settings load process, see https://github.com/stenzek/duckstation/blob/41cf93793dd0e6a0b11b1f1b68243b0149f4c943/src/core/settings.cpp#L225-L228